code/modules/mob/living/simple_animal/friendly/mouse.dm

```diff
@@ -91,6 +91,13 @@
 			visible_message("<span class='notice'>[src] nibbles through the [cheese], attracting another mouse!</span>")
 			qdel(cheese)
 			return
+	for(var/obj/item/reagent_containers/food/snacks/royalcheese/bigcheese in range(1, src))
+		visible_message("<span class='warning'>[src] devours [bigcheese]! He morphs into something... greater!</span>")
+		qdel(bigcheese)
+		qdel(src)
```

Comment:
see if you can transfer the ghost from the previous rat's body to the rat king's body if the mouse is ghost-controlled

otherwise, some shapeshift: mouse wizard somewhere is gonna die because they accidentally ate some nearby royale cheese.